### PR TITLE
Stupid Sexy Survivor: Synthetic Fabrics pt 2

### PR DESCRIPTION
**Pull request by tenmillimaster**
#### Summary
```SUMMARY: Content "Introduces synthetic fabrics as a material"```

#### Purpose of change
Previously, one could cut up a bunch of plastic clothing and make impermeable object out of them, such as canteens and jugs.

#### Describe the solution
2 new fabrics have been made: `lycra`, which is mostly nylon/polyethylene terephthalate(PET) mixed with polyurethane based elastomers, and `synthetic fabric`, a split between nylon/polyethylene terephthalate(PET).  Anywhere one would want stretchy, form fitting fabric, use `lycra`. Anywhere else where strong nylon or versatile PET is fine, use `synthetic fabric`.  Where impermeable plastics are used, like those in rain ponchos, rain coats, rubber boots, clean suits and hazmat suits, the original plastic will be used.

2 new materials requirement sets have been added: `"fabric_standard_permeable"`, for anything that you'd think of as 'normal clothing' like shirts, pants and the like, and `"fabric_standard_nostretch"` for any looser fitting clothing.

I have also added a number of sources for lycra, which are fun in and of themselves: unitards, leotards, and zentai (whole body stocking).  
I've added recipes for many of these have been added as well, mostly based on cotton counterparts. The overlooked belly holster has gotten a recipe too.

As replacing many of the 'rag' components of the recipes would be a GIANT pr, I've left that out for now.  No items or recipes touch 'synthetic fabric' just yet.
#### Describe alternatives you've considered
Not having a separate stretchy fabric. (stretches making compression garments from woven nylon though)

#### Testing
Loaded the game a couple times and spawned some of the items, compared the new items against the old.
#### Additional context

More chemically specific kinds of polymers assists in future recipes and such.

'Plastic' can now  largely be considered: LD/HD-Polyethylene, ABS plastic, PLA, and polypropylene. These are all more 'structural' plastics, good for making solid things.

'Snythetic fabric' can now broadly be considered to be woven nylon-type polymers, woven polyester(PET), and to a lesser extent, rayon.

This PR began life as a simple recipe to make deep concealment holsters.

This is a redo of #34864, after some git problems on my part.

### CODE REVIEW

File: data/json/items/armor.json [hunk 1]
```diff
@@ -5920,6 +5920,24 @@
     "material_thickness": 2,
     "flags": [ "VARSIZE", "WATER_FRIENDLY", "BELTED" ]
   },
+  {
+    "id": "leotard",
+    "type": "ARMOR",
+    "name": "leotard",
+    "name_plural": "leotard",
+    "description": "A skin-tight spandex garment, covering the body from the groin to the shoulders.   With its deep scoop-cut back and high waist cuts, this garment is perfect for showcasing your physique on the stage.",
```
Comment (Qrox) [inline data/json/items/armor.json]:
```suggestion
    "description": "A skin-tight spandex garment, covering the body from the groin to the shoulders.  With its deep scoop-cut back and high waist cuts, this garment is perfect for showcasing your physique on the stage.",
```

File: data/json/items/armor.json [hunk 2]
```diff
@@ -9610,5 +9645,39 @@
     "material_thickness": 1,
     "environmental_protection": 1,
     "flags": [ "OVERSIZE", "OUTER", "WATERPROOF", "ALLOWS_NATURAL_ATTACKS" ]
+  },
+  {
+    "id": "unitard",
+    "type": "ARMOR",
+    "name": "unitard",
+    "description": "A skin-tight spandex garment resembling a crew necked leotard combined with leggings.   Commonly used by gymnasts, dancers and equestrian vaulters, the unitard provides overall coverage with great flexibility.",
```
Comment (Qrox) [inline data/json/items/armor.json]:
```suggestion
    "description": "A skin-tight spandex garment resembling a crew necked leotard combined with leggings.  Commonly used by gymnasts, dancers and equestrian vaulters, the unitard provides overall coverage with great flexibility.",
```

File: data/json/items/armor.json (hunk 1 above)
Comment (tenmillimaster) [reply data/json/items/armor.json]: I am surprised the JSON style checks didn't catch these. Thanks.

**Comment by DMFan79** [comment]
Hi, the new items are dropping along with other loot but there's no way to recover lycra patches from them since they can be neither disassembled nor cut.

**Comment by tenmillimaster** [comment]
Thanks for the heads up, I'm looking into it now.

**Comment by Nerezza** [comment]
Was the leotard meant to cover the legs only for some reason?

**Comment by jbytheway** [comment]
Could you add / expand the comments in `material.json` to clarify the types of plastic they represent (similar to what you wrote in "Additional context" above)?

**Pull request merged**